tooling/nargo_cli/src/cli/check_cmd.rs

```diff
@@ -39,15 +37,18 @@ pub(crate) struct CheckCommand {
     show_program_hash: bool,
 }
 
-pub(crate) fn run(args: CheckCommand, config: NargoConfig) -> Result<(), CliError> {
-    let toml_path = get_package_manifest(&config.program_dir)?;
-    let selection = args.package_options.package_selection();
-    let workspace = resolve_workspace_from_toml(
-        &toml_path,
-        selection,
-        Some(NOIR_ARTIFACT_VERSION_STRING.to_string()),
-    )?;
+impl WorkspaceCommand for CheckCommand {
+    fn package_selection(&self) -> PackageSelection {
+        self.package_options.package_selection()
+    }
+    fn lock_type(&self) -> LockType {
+        // Creates a Nargo.toml template if it doesn't exist, otherwise only writes if `allow_overwrite` is true,
+        // so it shouldn't lead to any conflicts. Doesn't produce compilation artifacts.
+        LockType::None
```

Comment:
If `allow_overwrites` is true, wouldn't this need to lock the thing is writes?

Reply:
Ah, nevermind, I guess it writes the target and that wouldn't lead to any conflicts. I'm thinking if two check commands run in parallel, or if this command runs in parallel with another one that reads the target (I don't know if there's such command). So closing as resolved.